src/main/clojure/clara/rules/durability/fressian.clj

```diff
@@ -577,10 +580,13 @@
 
             forms-per-eval (or forms-per-eval com/forms-per-eval-default)
 
-            reconstruct-expressions (fn [ks]
+            reconstruct-expressions (fn [expr-lookup]
+                                      ;; Rebuilding the expr-lookup map from the serialized map:
+                                      ;; {[Int Keyword] {Keyword Any}} -> {[Int Keyword] [SExpr {Keyword Any}]}
                                       (into {}
-                                            (for [k ks]
-                                              [k (-> k meta (get (nth k 1)))])))
+                                            (for [[node-key compilation-ctx] expr-lookup]
+                                              [node-key [(-> compilation-ctx (get (nth node-key 1)))
+                                                         compilation-ctx]])))
```

Comment:
So the line and file info will be included with the serialized rulebase correct?  That will marginally increase the space consumed by the rulebase, but the size of sessions won't be impacted so I think this is OK.

Reply:
The rest of the `:compile-ctx` will be there too, such as `:condition` and `:production`. Previously this information was discarded after compiling the rulebase the first time. We then were making the assumption that it would recompile after deserialization. This way if for some reason a `fn`  cannot be recompiled, it should give better context to what condition/production the function was being compiled for.